.github/workflows/windows.yml

```diff
@@ -1,77 +1,147 @@
-        run:  |
-          export MSYSTEM_CARCH="$ARCH_${{ matrix.bits }}"
-          export MSYSTEM_CHOST="$ARCH_${{ matrix.bits }}-pc-msys"
-          ./scripts/build.sh --compiler ${{ matrix.compiler }} --build-type Debug --bin-path /mingw${{ matrix.bits }}/bin
-      - name:  Debug Warnings
-        run:   python scripts\count-warnings.py -m $env:MAX_WARNINGS_${{ matrix.compiler }}_${{ matrix.bits }}bit_Debug build.log
+          MAX_WARNINGS: ${{ matrix.conf.max_warnings }}
+        run:   python scripts\count-warnings.py build.log
+
+  build_windows_vs:
+    name: MSVC 32-bit
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        type: [Debug]
+    steps:
+      - uses:  actions/checkout@v1
+      - name:  Install packages
+        shell: pwsh
+        run: |
+          vcpkg install libpng sdl1 sdl1-net opusfile
+          vcpkg integrate install
+      - name:  Log environment
+        shell: pwsh
+        run:   .\scripts\log-env.ps1
+      - name:  Build
+        shell: pwsh
+        env:
+          PATH: '${env:PATH};C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\amd64'
+        run: |
+          cd vs
+          MSBuild -m dosbox.sln -p:Configuration=${{ matrix.type }}
+
+  build_windows_vs_release:
+    name: Release build
+    runs-on: windows-latest
+    steps:
+      - uses:  actions/checkout@v1
+      - name:  Install packages
+        shell: pwsh
+        run: |
+          vcpkg install libpng sdl1 sdl1-net opusfile
+          vcpkg integrate install
+      - name:  Log environment
+        shell: pwsh
+        run:   .\scripts\log-env.ps1
+      - name:  Inject version string
+        shell: bash
+        run: |
+          set -x
+          export VERSION=$(git describe --abbrev=4)
+          sed -i "s|VERSION \"git\"|VERSION \"$VERSION\"|" src/platform/visualc/config.h
+          echo ::set-env name=VERSION::$VERSION
+      - name:  Build
+        shell: pwsh
+        env:
+          PATH: '${env:PATH};C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\amd64'
+        run: |
+          cd vs
+          MSBuild -m dosbox.sln -p:Configuration=Release
+          # TODO "Windows SmartScreen" complains about unsigned .exe during
+          #      the first start of a build. Maybe self-signed certificate
+          #      will make it a bit more user-friendly.
+      - name:  Package
+        shell: bash
+        run: |
+          set -x
+          # Prepare content
+          mkdir -p dest/doc
+          cp vs/Release/dosbox.exe           dest/
+          cp COPYING                         dest/COPYING.txt
+          cp docs/README.template            dest/README.txt
+          cp docs/README.video               dest/doc/video.txt
+          cp README                          dest/doc/manual.txt
+          cp vs/Release/libpng16.dll         dest/
+          cp vs/Release/ogg.dll              dest/
+          cp vs/Release/opus.dll             dest/
```

Comment:
(question); is there also `opusfile.dll`? 
If not, then I suppose vcpkg archives all the symbols into one opus library.

Reply:
I checked and there's no `opusfile.dll`. When installing `opusfile`, vcpkg downloads `opus.dll`, `ogg.dll` and `opusfile.lib` - for static linking, I presume.

Reply:
ahh, great! I